capa/features/extractors/elf.py

```diff
@@ -501,7 +505,10 @@ def needed(self) -> Iterator[str]:
             if d_tag != DT_NEEDED:
                 continue
 
-            yield read_cstr(strtab, d_val)
+            try:
+                yield read_cstr(strtab, d_val)
+            except UnicodeDecodeError as e:
+                logger.warning("failed to read DT_NEEDED entry: %s", str(e))
```

Comment:
we should add more try/excepts around the parsing code, maybe even in/around `detect_elf_os` (best effort)